### PR TITLE
Support of P3 is added.

### PR DESCRIPTION
**Pull request by b0noI**

### CODE REVIEW

**Comment by sandeep-krishnamurthy** [comment]
1. Can you please update README file for these changes - what's new section - for new regions and instance types.
2. Can you add some details around the testing done for this changes?

**Comment by nswamy** [comment]
I think you also have to upload the .py file to S3 for it to fetch during bootstrap.

**Comment by lupesko** [comment]
@b0noI can you please follow up on the feedback so we can merge? thanks!

**Comment by b0noI** [comment]
@nswamy will do, immediately after I have LGTM on this PR and it is merged.

File: cfn-template/deeplearning.template [hunk 1]
```diff
@@ -127,10 +140,15 @@
...
       "us-west-2"      : { "URL" : "https://s3-us-west-2.amazonaws.com/" },
       "eu-west-1"      : { "URL" : "https://s3-eu-west-1.amazonaws.com/" },
-      "ap-southeast-2" : { "URL" : "https://s3-ap-southeast-2.amazonaws.com/" }
+      "us-east-2"      : { "URL" : "https://s3-us-east-2.amazonaws.com/" },
+      "ap-southeast-2" : { "URL" : "https://s3-ap-southeast-2.amazonaws.com/" },
+      "ap-northeast-1" : { "URL" : "https://s3-ap-northeast-1.amazonaws.com/" },
+      "ap-northeast-2" : { "URL" : "https://s3-ap-northeast-2.amazonaws.com/" },
+      "ap-south-1" : { "URL" : "https://s3-ap-south-1.amazonaws.com/" },
+      "eu-central-1" : { "URL" : "https://s3-eu-central-1.amazonaws.com/" },
+      "ap-southeast-1" : { "URL" : "https://s3-p-southeast-1.amazonaws.com/" }
```
Comment (sandeep-krishnamurthy) [inline cfn-template/deeplearning.template]: ap-southeast

**Comment by b0noI** [comment]
@sandeep-krishnamurthy requested changes are implemented, thank you for the review!

File: cfn-bootstrap/dl_cfn_setup.py [hunk 2]
```diff
@@ -47,7 +47,7 @@
 AWS_DL_DEFAULT_USER = None
 EFS_MOUNT = None
 
-AWS_GPU_INSTANCE_TYPES = [ "g2.2xlarge", "g2.8xlarge", "p2.xlarge", "p2.8xlarge", "p2.16xlarge" ]
+AWS_GPU_INSTANCE_TYPES = [ "g2.2xlarge", "g2.8xlarge", "p2.xlarge", "p2.8xlarge", "p2.16xlarge", "p3.2xlarge", "p3.8xlarge", "p3.16xlarge" ]
```
[the remark this answers is not in the document]
Comment (b0noI) [reply cfn-bootstrap/dl_cfn_setup.py]: you are right, changed to g3

**Comment by sandeep-krishnamurthy** [comment]
Thanks for the changes.
Can you please add details around Testing Done?

File: cfn-template/deeplearning.template (hunk 1 above)
Comment (b0noI) [reply cfn-template/deeplearning.template]: done, thank you!

**Comment by b0noI** [comment]
@sandeep-krishnamurthy I have created cluster and have executed distributed training with 2 nodes (1 master and 1 worker nodes)

**Comment by sandeep-krishnamurthy** [comment]
LGTM. @nswamy - If your comments are addressed, we can merge the change.
Thanks @b0noI

File: cfn-template/deeplearning.template [hunk 3]
```diff
@@ -150,7 +168,7 @@
         "Role": { "Fn::GetAtt" : ["LambdaExecutionRole", "Arn"] },
         "Code": {
             "S3Bucket":  {"Fn::Join" : ["", [{ "Ref" : "AWS::Region" }, { "Fn::FindInMap" : [ "Other", "S3SourceBucket", "BucketNameSuffix" ]} ] ]},
-            "S3Key": { "Fn::FindInMap" : [ "Other", "LambdaFunction", "FileName" ]},
+            "S3Key": { "Fn::FindInMap" : [ "Other", "LambdaFunction", "FileName" ]}
```
Comment (nswamy) [inline cfn-template/deeplearning.template]: i see you removed the `,` at the end of this line ? is this intentional?

Comment (b0noI) [reply cfn-template/deeplearning.template]: @nswamy yes, last object according to the json spec should not end with a coma (even though AWS parsers, looks like, are capable of ignoring such error)

**Pull request merged**